hax/hax/handler.py

```diff
@@ -276,6 +299,20 @@ def update_process_failure(self, planner: WorkPlanner,
                                 new_ha_states.append(
                                     HAState(fid=state.fid,
                                             status=current_status))
+
+                            # Cache the event in case of remote node failure
+                            # and if not already processed by RC node and
+                            # not updated in Consul KV.
+                            proc_status_kv = cns.get_process_status(state.fid)
+                            LOG.debug('proc status in kv=%s,'
+                                      'reported status=%s, node_fail=%s',
+                                      proc_status_kv.proc_status,
+                                      proc_status.name,
+                                      self._is_node_failure(state.fid))
```

Comment:
Let's just invoke this once and save it in a variable to avoid redundant processing.